frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx

```diff
@@ -1034,35 +1038,177 @@ export const ClusterServiceVersionDetails: React.SFC<ClusterServiceVersionDetail
+    catalogSource: CatalogSourceKind;
+    subscription: SubscriptionKind;
+    csv: ClusterServiceVersionKind;
+  },
+  boolean,
+  any,
+];
+
+const useCSVPageDataLoader = (dataLoaderProps: DataLoaderProps): DataLoaderResults => {
+  const { name, namespace } = dataLoaderProps;
+  const csvResource = {
+    kind: referenceForModel(ClusterServiceVersionModel),
+    name,
+    namespace,
+    isList: false,
+  };
+
+  const [csvData, csvLoaded, csvLoadError] = useK8sWatchResource<ClusterServiceVersionKind>(
+    csvResource,
+  );
+
+  const memoizedCSVData = useDeepCompareMemoize(csvData, true);
+
+  const operatorNamespace = memoizedCSVData?.metadata?.annotations?.['olm.operatorNamesapce'];
+
+  const subscriptionResource = {
+    isList: true,
+    kind: referenceForModel(SubscriptionModel),
+    namespace: operatorNamespace,
+  };
+
+  const [subscriptionData, subscriptionLoaded, subscriptionLoadError] = useK8sWatchResource<
+    SubscriptionKind[]
+  >(subscriptionResource);
+
+  const memoizedSubscriptionData = useDeepCompareMemoize(subscriptionData, true);
+
+  const currSubscription = React.useMemo(
+    () =>
+      _.find(memoizedSubscriptionData, {
+        status: {
+          installedCSV: name,
+        },
+      }),
+    [memoizedSubscriptionData, name],
+  );
+
+  const pkgManifestResource = {
+    kind: referenceForModel(PackageManifestModel),
+    name: currSubscription?.spec?.name,
+    namespace: currSubscription?.spec?.sourceNamespace,
+    isList: false,
+  };
+
+  const installPlanResource = {
+    kind: referenceForModel(InstallPlanModel),
+    namespace: currSubscription?.status?.installPlanRef.namespace,
+    name: currSubscription?.status?.installPlanRef.name,
+    isList: false,
+  };
+
+  const catalogSourceResource = {
+    kind: referenceForModel(CatalogSourceModel),
+    name: currSubscription?.spec?.source,
+    namespace: currSubscription?.spec?.sourceNamespace,
+    isList: false,
+  };
+
+  const data = useK8sWatchResources<{
+    packageManifest: PackageManifestKind;
+    installPlan: InstallPlanKind;
+    catalogSource: CatalogSourceKind;
+  }>({
+    packageManifest: pkgManifestResource,
+    installPlan: installPlanResource,
+    catalogSource: catalogSourceResource,
+  });
+
+  const {
+    packageManifest: {
+      data: pkgManifestData,
+      loaded: pkgManifestLoaded,
+      loadError: pkgManifestLoadError,
+    },
+    installPlan: {
+      data: installPlanData,
+      loaded: installPlanLoaded,
+      loadError: installPlanLoadError,
+    },
+    catalogSource: {
+      data: catalogSourceData,
+      loaded: catalogSourceLoaded,
+      loadError: catalogSourceLoadError,
+    },
+  } = data;
+
+  const loaded =
+    pkgManifestLoaded ||
+    installPlanLoaded ||
+    catalogSourceLoaded ||
+    csvLoaded ||
+    subscriptionLoaded;
+  const loadError =
+    pkgManifestLoadError ||
+    installPlanLoadError ||
+    catalogSourceLoadError ||
+    csvLoadError ||
+    subscriptionLoadError;
```

Comment:
Again, obsolete, but this would only show one loadError even if more than one failed.